crates/gpio/src/backend.rs

```diff
@@ -164,44 +193,24 @@ fn start_backend<D: 'static + GpioDevice + Send + Sync>(args: GpioArgs) -> Resul
             // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
             // trouble to other threads/guests or the main() function and should be safe for the
             // daemon.
-            let controller = match cfg {
+            // A separate thread is spawned for each socket and can connect to a separate guest.
+            // These are run in an infinite loop to not require the daemon to be restarted once a
+            // guest exits.
+            //
+            // There isn't much value in complicating code here to return an error from the
+            // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
+            // trouble to other threads/guests or the main() function and should be safe for the
+            // daemon.
+            match cfg {
                 GpioDeviceConfig::PhysicalDevice(device_num) => {
-                    let device = D::open(device_num).unwrap();
-                    GpioController::<D>::new(device).unwrap()
+                    let controller = PhysDevice::open(device_num).unwrap();
+                    start_device_backend(controller, socket.clone());
                 }
-                _ => {
-                    break;
+                GpioDeviceConfig::SimulatedDevice(num_gpios) => {
+                    let controller = DummyDevice::open(num_gpios).unwrap();
+                    start_device_backend(controller, socket.clone());
                 }
             };
```

Comment:
semicolon is unnecessary here I guess.

Reply:
See the last commit for an option to properly propagate errors  if things fail.